custom/icds_reports/static/js/directives/mobile-datepicker/mobile-datepicker.directive.js

```diff
@@ -339,18 +350,64 @@ angular.module('icdsApp').directive('mobileDatepicker', function() {
             }).on('WSlot.change',function(e,index){
                 initDate(parseInt($('.month').WSlot('get')),parseInt($('.year').WSlot('getText')),$('.day').WSlot('get'));
                 updateText();
+                let center = $scope.date.getMonth();
+                if ($scope.date.getFullYear() === 2017) {
```

Comment:
can we reduce the duplication of some of this logic from the initial setup and the event handler by creating more helper functions?

Reply:
sure